inference-engine/src/mkldnn_plugin/nodes/mkldnn_reorder_node.h

```diff
@@ -20,11 +20,20 @@ class MKLDNNReorderNode : public MKLDNNNode {
 
     void getSupportedDescriptors() override;
     void initSupportedPrimitiveDescriptors() override;
-    void createPrimitive() override;
     void execute(mkldnn::stream strm) override;
     bool created() const override;
     const std::vector<impl_desc_type>& getPrimitivesPriority() override;
 
+    void createPrimitive() override;
+
+    std::vector<VectorDims> shapeInfer() const override {
+        return {getParentEdgesAtPort(0)[0]->getMemory().getStaticDims()};
+    }
```

Comment:
Please move to cpp. Also do not forget that the reorder node also support shape mismatch cases in specific conditions.

Reply:
as far as I understand we use this behavior for group convolution weight when we need to reshape `OIHW` -> `GOIHW` (I suppose that after ngraph migration we no need this WA). But group convolution weight for dynamic cases always will be `Constant` so we no need `shapeInfer` for this case

At this moment I have no idea how we can provide generic decision for this cases.
How we can understand how we should perform reshaping?
i.e. we have defined `src dims`: [2, 4, 6, 8] with rank == 4 and some `dst dims` with rank == 5 [?, ?, ?, ?, ?]
we can can perform reshape next way:
1. [1, 2, 4, 6, 8]
2. [2, 2, 2, 6, 8]
3. [2, 4, 2, 3, 8]
etc.

so my suggestion: throw exception if `src` and `dst` rank mismatch.

Correct me please if I missed something and let's discuss it offline.